### PR TITLE
⚙️Configure with qawolf.config.js

### LINKED ISSUE #353: ✏️Support typescript in `npx qawolf create`

### PR DESCRIPTION
**Pull request by jperl**
Resolves #353
Resolves part of #515

To Do

- [x] useTypeScript in qawolf.config.js. emit declarations and make exports typescript friendly.
- [x] use rootDir from qawolf.config.js, so users can specify their test directory (ex. rootDir: "tests/e2e")
- [x] generate script/tests in rootDir so it can be a top level directory, keep selectors a subdirectory. rootDir/myTest.test.js, rootDir/myScript.js, rootDir/selectors/myTest.json
- [x] attribute in qawolf.config.js
- [x] createScriptTemplate, createTestTemplate: ({ device, isTypeScript, name, statePath, url }) => string in qawolf.config.js
- [x] npx qawolf create: set process.env.QAW_CREATE=true
- [x] add testTimeout to config, use for non-repl

Example `qawolf.config.js`

```
module.exports = {
  // jest --config
  config: '{}',
  createScriptTemplate:  ({ name, url }) => 'myTemplate',
  createTestTemplate:  ({ name, url }) => 'myTemplate',
  // where to create scripts/tests
  rootDir: ".qawolf",
  // jest --testTimeout
  testTimeout: 60000,
  useTypeScript: true
};
```

### CODE REVIEW

**Comment by jperl** [comment]
Example `qawolf.config.js`

```
const createTestTemplate = ({ name, url }) =>
  `const qawolf = require('qawolf');
const selectors = require('./selectors/${name}.json');

let browser;
let page;

beforeAll(async () => {
  browser = await qawolf.launch();
  const context = await browser.newContext();
  await qawolf.register(context);
  page = await context.newPage();
});

afterAll(async () => {
  await qawolf.stopVideos();
  await browser.close();
});

test('${name}', async () => {
  // MY SPECIAL SETUP
  await page.goto('${url}');
  await qawolf.create();
});`;

module.exports = {
  createTestTemplate,
  isTypeScript: true,
  rootDir: "qawolf"
};
```

**Comment by jperl** [comment]
Separately we need to update docs (install/typescript).

File: README.md [hunk 1]
```diff
@@ -101,7 +101,7 @@ Create a [Playwright](https://github.com/microsoft/playwright) and [Jest](https:
 npx qawolf create url [name]
 ```
 
-Edit your test code as it is created by opening `.qawolf/tests/myTestName.test.js`.
+Edit your test code as it is created by opening `.qawolf/myTestName.test.js`.
```
Comment (flaurida) [inline README.md]: is it `.qawolf` or `qawolf`? just checking

Comment (jperl) [reply README.md]: I think we should default to `.qawolf` for backwards compatibility?

**Pull request merged**